components/integrations/edit_outgoing_webhook/edit_outgoing_webhook.tsx

```diff
@@ -14,105 +16,110 @@ const HEADER = {id: 'integrations.edit', defaultMessage: 'Edit'};
+    enablePostUsernameOverride: boolean;
+
+    /**
+     * Whether to allow configuration of the default post icon.
+     */
+    enablePostIconOverride: boolean;
+}
 
-    constructor(props) {
-        super(props);
+interface State {
+    showConfirmModal: boolean;
+    serverError: string;
+}
 
+export default class EditOutgoingWebhook extends React.PureComponent<Props, State> {
+    private newHook: OutgoingWebhook | undefined;
+
+    constructor(props: Props) {
+        super(props);
         this.state = {
             showConfirmModal: false,
             serverError: '',
         };
     }
 
-    componentDidMount() {
+    componentDidMount(): void {
         if (this.props.enableOutgoingWebhooks) {
             this.props.actions.getOutgoingHook(this.props.hookId);
         }
     }
 
-    editOutgoingHook = async (hook) => {
+    editOutgoingHook = async (hook: OutgoingWebhook): Promise<void> => {
         this.newHook = hook;
 
-        if (this.props.hook.id) {
-            hook.id = this.props.hook.id;
+        if (this.props.hook!.id) {
+            hook.id = this.props.hook!.id;
         }
 
-        if (this.props.hook.token) {
-            hook.token = this.props.hook.token;
+        if (this.props.hook!.token) {
+            hook.token = this.props.hook!.token;
         }
 
-        const triggerWordsSame = (this.props.hook.trigger_words.length === hook.trigger_words.length) &&
-            this.props.hook.trigger_words.every((v, i) => v === hook.trigger_words[i]);
+        const triggerWordsSame = (this.props.hook!.trigger_words.length === hook!.trigger_words.length) &&
+            this.props.hook!.trigger_words.every((v, i) => v === hook.trigger_words[i]);
 
-        const callbackUrlsSame = (this.props.hook.callback_urls.length === hook.callback_urls.length) &&
-            this.props.hook.callback_urls.every((v, i) => v === hook.callback_urls[i]);
+        const callbackUrlsSame = (this.props.hook!.callback_urls.length === hook!.callback_urls.length) &&
+            this.props.hook!.callback_urls.every((v, i) => v === hook.callback_urls[i]);
 
-        if (this.props.hook.content_type !== hook.content_type ||
+        if (this.props.hook!.content_type !== hook.content_type ||
             !triggerWordsSame || !callbackUrlsSame) {
             this.handleConfirmModal();
         } else {
             await this.submitHook();
         }
     }
 
-    handleConfirmModal = () => {
+    handleConfirmModal = (): void => {
         this.setState({showConfirmModal: true});
     }
 
-    confirmModalDismissed = () => {
+    confirmModalDismissed = (): void => {
         this.setState({showConfirmModal: false});
     }
 
-    submitHook = async () => {
+    submitHook = async (): Promise<void> => {
         this.setState({serverError: ''});
 
-        const {data, error} = await this.props.actions.updateOutgoingHook(this.newHook);
+        const {data, error}: any = await this.props.actions.updateOutgoingHook(this.newHook!);
```

Comment:
Is it possible to type this as `{data: OutgoingWebhook, error: ServerError}`?

Reply:
This result in the following error

```
TS2739: Type 'ActionFunc' is missing the following properties from type '{ data: OutgoingWebhook; error: ServerError; }': data, error
```
the `ActionFunc` returns `ActionResult` and the type definition is this
```
export type ActionResult = {
    data: any;
} | {
    error: any;
};
```

Reply:
So what you can do here is use the `ActionCreatorsMapObject` in the redux connector to manually define the return type. Usually that will fix this.

Reply:
I'm sorry I didn't get that. Could you elaborate or connect with me in the mattermost developer community for a brief?

Reply:
Sent a message in MM :)